src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -178,6 +178,9 @@ async def generate_python_from_protobuf(
         Process(
             argv,
             input_digest=input_digest,
+            immutable_input_digests={
+                protoc_relpath: downloaded_protoc_binary.digest,
+            },
```

Comment:
Consider exposing `immutable_inputs_digests` as a property of `DownloadedExternalTool`, similar to https://github.com/pantsbuild/pants/blob/6f85d2579bb96f40d106659e956d3c8dceeb0915/src/python/pants/jvm/resolve/coursier_setup.py#L132-L134? Could add `immutable_inputs_exe` too.

Reply:
I love this idea!